### ISSUE TITLE
Maven repository has javax as dependency for <3.15 versions but documentation uses jakarta

### ISSUE THREAD

**Issue by 18anandn** [jOOQ/jOOQ#17503]
### The documentation URL

https://www.jooq.org/doc/3.15/manual/code-generation/codegen-jpa/

### Feedback

[Maven artifact 3.15](https://mvnrepository.com/artifact/org.jooq/jooq/3.15.12).

Is there even a official or unofficial codegen plugin that supports codegen from javax annotated entities?

**Comment by lukaseder** [comment]
> Maven repository has javax as dependency for <3.15 versions but documentation uses jakarta

Interesting, we're generating only `jakarta` imports for all manual versions. I guess that could be fixed. It's correct elsewhere, e.g.:

- `javax.*`: https://www.jooq.org/doc/3.15/manual/code-generation/codegen-advanced/codegen-config-generate/codegen-generate-annotations/
- `jakarta.*`: https://www.jooq.org/doc/3.16/manual/code-generation/codegen-advanced/codegen-config-generate/codegen-generate-annotations/

> Is there even a official or unofficial codegen plugin that supports codegen from javax annotated entities?

You can probably continue using the 3.15 version of the `JPADatabase` - have you tried that? Alternatively, the `JPADatabase` is just a few lines of code, which you can copy paste and adapt as needed.

**Comment by 18anandn** [comment]
@lukaseder Thank you for the super quick response!

I followed the implementation suggested on [this](https://www.jooq.org/doc/3.15/manual/code-generation/codegen-jpa/) page for doing codegen using JPA annotated entities. I put the entities in a different module than the "jooq" module (which will do the codegen). Unfortunately, it still gives me the message that either there is no package by the given name or I haven't set up the dependencies correctly. I made sure that the dependencies are correct by checking the dependency tree and I have set the right package name (it's a multi-project setup).

According to [this](https://github.com/etiennestuder/gradle-jooq-plugin/issues/570#issuecomment-1334533591) comment, it says that JPADatabase uses jakarta annotations instead of javax to detect entities. Maybe I will try your suggestion of copy pasting and modifying it. Thanks!

PS: I am using gradle groovy with Kotlin.
